### ISSUE TITLE
Migrate `wp_enqueue_style_block()` content to including CSS and JavaScript page

### ISSUE THREAD

**Issue by justintadlock** [WordPress/Documentation-Issue-Tracker#451]
## Issue Description

The Including CSS and JS page in the theme handbook does not currently have documentation on loading per-block stylesheets via `wp_enqueue_block_style()`.

## URL of the Page with the Issue

https://developer.wordpress.org/themes/basics/including-css-javascript/

## Why is this a problem?

With WordPress' ability to only load CSS for blocks on an as-needed basis, we should be pushing theme authors to do the same whenever custom CSS is required beyond what is offered through core or `theme.json`-based methods.  The first option should almost always be `wp_enqueue_block_style()` if the CSS is related to a block.

## Suggested Fix

Add a new section describing both _why_ per-block styles are a good thing and _how_ to use them in themes.

**Comment by carolinan** [comment]
It does, but it is not in the basic section it is in the block theme section.
https://developer.wordpress.org/themes/block-themes/block-theme-setup/#including-css-for-block-styles

**Comment by carolinan** [comment]
I think the "including CSS and Javascript" page needs to be rewritten / re-organised, my main concern is that it is not clear what works for what type of theme.

-There are lots of questions in the form of "Why isn't my JS loading in the site editor" and similar.

- Another concern of mine is that while reading this page, the reader has only briefly been introduced to the concept of there being more than one type of theme. So for example if this page would say "You only need to enqueue the comment reply script in a classic theme", they may be confused, asking why.

**Comment by DaisyOlsen** [comment]
I've given this a closer look, and I tend to agree with @carolinan this page could use an overhaul.  I will update the title of this issue to reflect that as the task I aim to tackle.

I've got a draft Google Doc that I'm busily marking up. If anyone wants to participate, I would be thrilled to pair up.

**Comment by DaisyOlsen** [comment]
Edit... I **would** update the issue title if I had triage access.

**Comment by justintadlock** [comment]
For this specific ticket, I want to make sure we're focused on the original topic. My initial goal was to add documentation on per-block stylesheets. Technically, [this documentation already exists](https://developer.wordpress.org/themes/block-themes/block-theme-setup/#including-css-for-block-styles). However, it is in a location in the handbook that I wouldn't think to look for it.

To resolve this ticket, I would choose between one of the following options:

- Add a link from the Including CSS and JS page to the current documentation section.
- Migrate/Copy the current documentation section to the Including CSS and JS page.

This doesn't solve the larger issue with the need for a rewrite/reorganization. However, it does help with discoverability issue that the ticket was meant to solve. Let's go for the smaller win for the moment.

For the longer term, we should have another ticket for reorganizing the CSS and JS page (I'd like to propose some bigger changes soon, and I'll have that proposal ready in the near future).

**Comment by carolinan** [comment]
Move it to Including CSS and JS.

**Comment by DaisyOlsen** [comment]
Unless I hear otherwise, I'll make the adjustments directly on the site and schedule the revision for a future date to allow the content to be reviewed before it goes live. If anyone prefers a different process, just let me know. This should be a very quick adjustment.

**Comment by DaisyOlsen** [comment]
@carolinan I've scheduled the updates to happen one week from now.

The indicated content was moved to this page just below the stylesheets section: 

https://developer.wordpress.org/wp-admin/post.php?post=144992&action=edit

Original content has been changed to a link to refer to the new location:

https://developer.wordpress.org/wp-admin/post.php?post=144994&action=edit

The only changes I made were to rephrase the opening sentence which linked to the page its now on and to fix some `code` formatting.

If the changes look good, feel free to change the publish date to something sooner. This seemed the easiest way to handle seeing the changes in place for review.
